programs/ownable_api/src/ownable_instruction.rs

```diff
@@ -0,0 +1,68 @@
+use num_derive::{FromPrimitive, ToPrimitive};
+use serde_derive::{Deserialize, Serialize};
+use solana_sdk::{
+    instruction::{AccountMeta, Instruction},
+    instruction_processor_utils::DecodeError,
+    pubkey::Pubkey,
+    system_instruction,
+};
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, FromPrimitive, ToPrimitive)]
+pub enum OwnableError {
+    IncorrectOwner,
+}
+
+impl<T> DecodeError<T> for OwnableError {
+    fn type_of() -> &'static str {
+        "OwnableError"
+    }
+}
+
+impl std::fmt::Display for OwnableError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(
+            f,
+            "{}",
+            match self {
+                OwnableError::IncorrectOwner => "incorrect owner",
+            }
+        )
+    }
+}
+impl std::error::Error for OwnableError {}
+
+fn initialize_account(contract_pubkey: &Pubkey, owner_pubkey: &Pubkey) -> Instruction {
+    let keys = vec![AccountMeta::new(*contract_pubkey, false)];
+    Instruction::new(crate::id(), &owner_pubkey, keys)
+}
+
+pub fn create_account(
+    payer_pubkey: &Pubkey,
+    contract_pubkey: &Pubkey,
```

Comment:
`contract_pubkey `-> `account_pubkey`?

Reply:
When processing the instruction you refer to this pubkey as `account_pubkey` but when creating the instruction and testing you call it `contract_pubkey`

Reply:
Fixed, thanks!